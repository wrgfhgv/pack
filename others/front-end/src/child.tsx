import { useContext } from 'react';
import { MyContext } from './index';
export function Child() {
  const { value } = useContext(MyContext) as any;
  return <div>{value}</div>;
}
