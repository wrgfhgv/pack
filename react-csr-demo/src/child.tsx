import React, { Suspense } from 'react';
import './style.css';

export type RefType = {
    sayHello: () => void;
}

const ChildComponent = (props: any) => {
   console.log(props.ref);
   
    return (
        <Suspense>
            {props.name}
            <h1 className='box1'>Child</h1>
        </Suspense>
    )
}

export const Child = ChildComponent;