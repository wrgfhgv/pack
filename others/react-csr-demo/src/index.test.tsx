import React from 'react';
import { Child } from './child';
import { render } from '@testing-library/react';

test('child component should render', () => {
    const { asFragment } = render(<Child name={'test'}/>);
    expect(asFragment()).toMatchSnapshot();
});
