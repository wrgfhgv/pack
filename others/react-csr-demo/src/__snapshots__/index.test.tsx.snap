// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`child component should render 1`] = `
<DocumentFragment>
  <div>
    test
    <h1>
      Child
    </h1>
  </div>
</DocumentFragment>
`;
